draft-ietf-quic-transport.md

```diff
@@ -865,92 +865,85 @@ endpoint, as described in {{termination}}.
+
+If the server supports the version, the server operates in accordance with
+the specification of that version. For the version described in this
+specification, it checks if the packet is a correctly formatted Initial packet.
+If so, the server MUST proceed with the handshake ({{handshake}}).  This
+commits the server to the version that the client selected. If not an Initial
+packet, the server MUST either buffer the packet or silently drop it.
+
+### Handshake Buffering {{#handshake-buffer}}
+
+Due to packet reordering or loss, subsequent packets for a handshake might
+arrive at the server prior to the intended Initial packet. As described above,
+servers MAY buffer these packets in anticipation of the Initial packet
+arriving later. This is especially useful for 0-RTT packets that routinely
+accompany Initial packets.
 
 Buffering ensures that data is not lost, which improves performance; conversely,
 discarding these packets could create false loss signals for the congestion
 controllers.  However, limiting the number and size of buffered packets might be
 needed to prevent exposure to denial of service.
 
-For clients, any packet that cannot be associated with an existing connection
-SHOULD be discarded if it is not buffered.  Discarded packets MAY be logged for
-diagnostic or security purposes.
+Servers MUST NOT send packets in response to these buffered packets until
+the initial packet arrives.
 
-For servers, packets that aren't associated with a connection potentially create
-a new connection.  However, only packets that use the long packet header and
-that are at least the minimum size defined for the protocol version can be
-initial packets.  A server MAY discard packets with a short header or packets
-that are smaller than the smallest minimum size for any version that the server
-supports.  A server that discards a packet that cannot be associated with a
-connection MAY also generate a stateless reset ({{stateless-reset}}).
+In this version of QUIC, clients cannot send short form headers prior to a
+response from the server. Servers that do not support a version of QUIC that
+allows early short headers MUST either respond with a stateless reset or drop
+the packet silently.
 
-This version of QUIC defines a minimum size for initial packets of 1200 octets
-(see {{packetization}}).  Versions of QUIC that define smaller minimum initial
-packet sizes need to be aware that initial packets will be discarded without
-action by servers that only support versions with larger minimums.  Clients that
-support multiple QUIC versions can avoid this problem by ensuring that they
-increase the size of their initial packets to the largest minimum size across
-all of the QUIC versions they support.  Servers need to recognize initial
-packets that are the minimum size of all QUIC versions they support.
+## Version Negotiation
 
+{{packet-handling}} describes the conditions under which endpoints negotiate
+versions.
```

Comment:
I think that we could keep some of the exposition here.  Maybe: "Version negotiation ensures that client and server agree to a QUIC version that is mutually supported.  A server sends a Version Negotiation packet in response to a packet that might initiate a new connection, see {{packet-handling}} for details."